src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -504,5 +513,33 @@ private static string[] SplitFlagsEnum(string value)
 #endif
                 );
         }
+
+        private static bool ContainsSpecialChar()
+        {
+#if NETCOREAPP
+            string[] names = Enum.GetNames<T>();
+#else
+            string[] names = Enum.GetNames(typeof(T));
+#endif
```

Comment:
I think using the same API is fine here:
```suggestion
            string[] names = Enum.GetNames(typeof(T));
```

Reply:
Using the generic overload does or will in the future help in AOT scenarios.

Reply:
Ah fair, I assumed one method was calling the other but it seems we're special casing AOT

https://github.com/dotnet/runtime/blob/c6c8bb8b73ff4f7a686ca5a350fbb0093b3b708b/src/libraries/System.Private.CoreLib/src/System/Enum.cs#L325-L328